src/tooltips.cpp

```diff
@@ -60,16 +60,56 @@ void tooltip::init_label()
 	rect game_canvas = video::game_canvas();
 	unsigned int border = 10;
 
+	bool mydebug=false;
+
+	rect huge;
+	huge.h=1000000;
+	huge.w=1000000;
+
 	label.set_font_size(font_size);
 	label.set_color(font::NORMAL_COLOR);
-	label.set_clip_rect(game_canvas);
-	label.set_width(text_width);
+	label.set_clip_rect(huge);  
+	label.set_width(text_width);   // If tooltip will be too tall for game_canvas, this could be scaled up appropriately
 	label.set_alignment(font::LEFT_ALIGN);
 	label.set_bg_color(bgcolor);
 	label.set_border_size(border);
 
 	label.create_texture();
 
+	point lsize = label.get_draw_size();
+	int new_text_width = text_width;
+	if(lsize.y > game_canvas.h*0.95) {
+		// The multiplier (currently 2) is hard to select, since making a texture wider is no guarantee that there will be fewer lines of text.
+		// Could just use something really big, or incrementally increase it until the texture fits vertically (or width maxes out)?
+		//
+		// This block of text is just
+		// as tall as the other one.
+		//
+		// This block of text is just as tall as the other
+		// one.
+		new_text_width = 2 * text_width * static_cast<float>(static_cast<float>(lsize.y)/game_canvas.h);
+		if(new_text_width>game_canvas.w) {
+			new_text_width=game_canvas.w;
+		}
+	}
+	if(mydebug) {LOG_FT << "lsize.x,y = " << lsize.x << "," << lsize.y << ", new_text_width = " << new_text_width;}
+
+
+//	label.clear_texture();  // couldn't figure this out  -- just commented out the check if tex_ exists in create_texture()
+
+	label.set_font_size(font_size);
+	label.set_color(font::NORMAL_COLOR);
+	label.set_clip_rect(game_canvas);  
+	label.set_width(new_text_width);   
+	label.set_alignment(font::LEFT_ALIGN);
+	label.set_bg_color(bgcolor);
+	label.set_border_size(border);
```

Comment:
Seems like duplicate of lines 69 to 75.

Reply:
They are, almost.  I couldn't figure out how to get the height of the text (to calculate the width for the label) without building the label, so I end up building it twice (see "new_text_width" line 103).

I think the answer is to build a pango layout from the text and use pango_layout_get_pixel_size to get the tentative dimensions, but that's beyond my ability right now.

Reply:
What I'm saying is that you don't need to set alignment/color/fontsize etc. twice. You've already set them once, so that should work. Try commenting out lines 69 to 77 (including the create_texture call at line 77) and recompiling it to see if it works the same. Those lines are redudant since you're recreating the texture at line 108, so the attribute setting and the created texture at line 77 does nothing.
(If it doesn't work the same after commenting out those lines, you're free to keep them.)

Reply:
Ignore lines 71 and 72 that you added.

Reply:
I see